src/Language/Visitor.php

```diff
@@ -417,18 +417,20 @@ public static function visitInParallel($visitors)
 
                     $result = $fn(...func_get_args());
 
-                    if ($result instanceof VisitorOperation) {
-                        if ($result->doContinue) {
-                            $skipping[$i] = $node;
-                        } elseif ($result->doBreak) {
-                            $skipping[$i] = $result;
-                        } elseif ($result->removeNode) {
-                            return $result;
-                        }
-                    } elseif ($result !== null) {
+                    if (! ($result instanceof VisitorOperation)) {
+                        continue;
+                    }
+
+                    if ($result->doContinue) {
+                        $skipping[$i] = $node;
+                    } elseif ($result->doBreak) {
+                        $skipping[$i] = $result;
+                    } elseif ($result->removeNode) {
                         return $result;
```

Comment:
Looks like we miss this branch now? When the result is not an instance of `VisitorOperation` we must return it

Reply:
Sorry, what is your suggestion?

Reply:
I think I understand what you mean. With your skip change, that bit of code was being flagged as dead by the coverage tests, so I removed it.

Reply:
GitHub shows this comment against new code but I've actually left it against the old code. So I mean this part:

```php
elseif ($result !== null) {
    return $result;
}
```

is missing now. We must restore it as it is basically how you "edit" an AST node with a parallel visitor.

Reply:
Restored in 11b59cd. Are you sure this is necessary? I only ask because the tests passed without it.

Reply:
Absolutely. Probably missing tests for this (we don't use a parallel visitor to modify nodes in this library, yet this feature is indispensable for AST manipulations)

Reply:
Actually, can we revert this change completely (except for `$visitFn !== null` part)? Doesn't look more readable to me after the change.

Reply:
Wait, what? Revert what change? The one I just made?

Reply:
I mean all changes in this file except for `if ($visitFn !== null) {` line. Those changes don't really change anything now that we restored `return $result`. Or am I missing something?

Reply:
Sure, you got it 7e680fd